app/views/shared/_run_table.slim

```diff
@@ -29,7 +32,20 @@
             - if cols.include?(:runner)
               td.align-right = user_badge(run.user)
             - if cols.include?(:time)
-              td.align-right = run.duration.format
+              td.align-right
+                / Make sure to not group runs with no category
+                - if col_options[:time].include?(:archived) && run.category_id.present?
+                  / Gather only runs with the same category and sort by default timing ascending
+                  - old_runs = all_runs.filter { |r| r.category_id == run.category_id }
+                  - old_runs = old_runs.sort_by(&Run.duration_type(run.default_timing))
```

Comment:
Now that we have `User#old_runs_for_page`, if we have that return a hash rather than an array, we can migrate some of this logic out of the view (and avoid the n `filter` calls as a bonus).